qcodes/instrument_drivers/Keysight/N9030B.py

```diff
@@ -0,0 +1,601 @@
+        stop = self.stop()
+        if val >= stop:
+            raise ValueError(f"Start frequency must be smaller than stop "
+                             f"frequency. Provided start freq is: {val} Hz and "
+                             f"set stop freq is: {stop} Hz")
+
+        self.write(f":SENSe:FREQuency:STARt {val}")
+
+        start = self.start()
+        if abs(val - start) >= 1:
+            self.log.warning(
+                f"Could not set start to {val} setting it to {start}"
+            )
+
+    def _set_stop(self, val: float) -> None:
+        """
+        Sets stop frequency
+        """
+        start = self.start()
+        if val <= start:
+            raise ValueError(f"Stop frequency must be larger than start "
+                             f"frequency. Provided stop freq is: {val} Hz and "
+                             f"set start freq is: {start} Hz")
+
+        self.write(f":SENSe:FREQuency:STOP {val}")
+
+        stop = self.stop()
+        if abs(val - stop) >= 1:
+            self.log.warning(
+                f"Could not set stop to {val} setting it to {stop}"
+            )
+
+    def _set_center(self, val: float) -> None:
+        """
+        Sets center frequency and updates start and stop frequencies if they
+        change.
+        """
+        self.write(f":SENSe:FREQuency:CENTer {val}")
+        self.update_trace()
+
+    def _set_span(self, val: float) -> None:
+        """
+        Sets frequency span and updates start and stop frequencies if they
+        change.
+        """
+        self.write(f":SENSe:FREQuency:SPAN {val}")
+        self.update_trace()
+
+    def _set_npts(self, val: int) -> None:
+        """
+        Sets number of points for sweep
+        """
+        self.write(f":SENSe:SWEep:POINts {val}")
+
+    def _enable_auto_sweep_time(self, val: str) -> None:
+        """
+        Enables auto sweep time
+        """
+        self.write(f":SENSe:SWEep:TIME:AUTO {val}")
+
+    def _enable_auto_sweep_type(self, val: str) -> None:
+        """
+        Enables auto sweep type
+        """
+        self.write(f":SENSe:SWEep:TYPE:AUTO {val}")
+
+    def _set_sweep_type(self, val: str) -> None:
+        """
+        Sets sweep type
+        """
+        self.write(f":SENSe:SWEep:TYPE {val}")
+
+    def _get_data(self, trace_num: int) -> ParamRawDataType:
+        """
+        Gets data from the measurement.
+        """
+        try:
+            timeout = self.sweep_time() + self.root_instrument._additional_wait
+            with self.root_instrument.timeout.set_to(timeout):
+                data_str = self.ask(f":READ:"
+                                    f"{self.root_instrument.measurement()}"
+                                    f"{trace_num}?")
+                data = np.array(data_str.rstrip().split(",")).astype("float64")
+        except TimeoutError as e:
+            raise RuntimeError("Couldn't receive any data. Command timed "
+                               "out.") from e
```

Comment:
Why re throw the TimeoutError as a RunTimeError. It would seem better to Re Raise this as a TimeOutError